src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -1494,52 +1488,51 @@ internal Type[] FunctionPointerReturnAndParameterTypes
                 }
             }
 
-            internal string? GetName(TypeNameKind kind)
+            // No namespace, full instantiation, and assembly.
+            internal string GetName() => m_name ?? ConstructName(ref m_name, TypeNameFormatFlags.FormatBasic)!;
+
+            // No assembly.
+            internal string? GetFullName() => m_fullName ??
+                (IsFullNameRoundtripCompatible(m_runtimeType)
+                    ? ConstructName(ref m_fullName, TypeNameFormatFlags.FormatNamespace | TypeNameFormatFlags.FormatFullInst)
```

Comment:
Nit: The assignment in ConstructName may be changed to just `=`. Also, we may want to mark ConstructName with NoInlining to make it clear to the JIT that it is a slow path that's not worth inlining.